From 92b524db3114a00f0f28e52c5f4049dd323e80d3 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 13 Jun 2022 12:42:48 +0300
Subject: [PATCH] Revert "[SW-82717] rom: add prot bits programming"

This reverts commit d7ded5c2c53024e4196915045a6fbf87cb42c719.

As per the updated security specification, secured firmware support
is not supported in rom recovery flow. Hence the corresponding
protection bit programming can be removed.

Change-Id: I17f82360e1dab97f0bdb00c6c6d6df707ac25585
---

diff --git a/arch/gaudi3/arc/dev_security.h b/arch/gaudi3/arc/dev_security.h
index 607107e..04fa2ef 100644
--- a/arch/gaudi3/arc/dev_security.h
+++ b/arch/gaudi3/arc/dev_security.h
@@ -26,18 +26,6 @@
 				KEY_PROP_EXP_SZ + KEY_PROP_INV_SZ + \
 				KEY_PROP_NB_SZ)
 
-/*
- * PSOC_GLOBAL_CONF protection bit register position in SPECAIL_GLBL_PRIV
- * which require user defined configuration value
- */
-enum prot_bits_reg_idx {
-	BTM_FSM_PRSTN_REG_IDX = 0,
-	BTL_ROM_STATUS_REG_IDX = 1,
-	SCRATCHPAD_REG_IDX = 2,
-	KMD_MSG_TO_CPU_REG_IDX = 6,
-	PROT_BITS_MAX_CONF,
-};
-
 /* 784 + 8 = 792 */
 #define TOTAL_PUB_KEY_PROPS_SZ  (PUB_KEY_SZ + KEY_PROP_HDR_SZ)
 
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 42e61f0..ca63704 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -816,37 +816,6 @@
 	security_rr_config_hbw_if_common(&rr_ctx);
 }
 
-#define SPECIAL_PRIV_PROTECT_ALL	0x0
-#define SPECIAL_PRIV_OPEN_ALL		0xFFFFFFFF
-#define MAX_NUM_SPECIAL_PRIV_REGS	32
-#define SPECIAL_GLBL_SEC_IDX		30
-static uint32_t prot_bits_user_config[PROT_BITS_MAX_CONF] = {
-	0x20400,
-	0x8,
-	SPECIAL_PRIV_OPEN_ALL,
-	SPECIAL_PRIV_PROTECT_ALL,
-	SPECIAL_PRIV_PROTECT_ALL,
-	SPECIAL_PRIV_PROTECT_ALL,
-	0x3,
-};
-
-void security_prot_bits_config(void)
-{
-	int i, reg_addr;
-	for (i = 0; i < MAX_NUM_SPECIAL_PRIV_REGS; i++)
-	{
-		reg_addr = mmD0_PSOC_GLOBAL_CONF_SPECIAL_GLBL_PRIV_0 + i * sizeof(uint32_t);
-
-		if (i < PROT_BITS_MAX_CONF)
-			regmap_writel(prot_bits_user_config[i], reg_addr);
-		else if (i == SPECIAL_GLBL_SEC_IDX)
-			/* TODO: Check if explicit write to sec reg is required */
-			regmap_writel(SPECIAL_PRIV_OPEN_ALL, reg_addr);
-		else
-			regmap_writel(SPECIAL_PRIV_PROTECT_ALL, reg_addr);
-	}
-}
-
 static void security_rr_recovery_unsecured_config(void)
 {
 	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
@@ -865,7 +834,6 @@
 void security_rr_recovery_config(void)
 {
 	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK) {
-		security_prot_bits_config();
 		security_rr_recovery_lbw_config();
 		security_rr_recovery_hbw_config();
 	} else {
